ethereum/networks/src/main/java/tech/pegasys/teku/networks/Eth2NetworkConfiguration.java

```diff
@@ -146,8 +146,14 @@ public String toString() {
     public Eth2NetworkConfiguration build() {
       checkNotNull(constants, "Missing constants");
 
+      final Spec spec = SpecFactory.create(constants);
+      // if the deposit contract was not set, default from constants
+      if (eth1DepositContractAddress.isEmpty()) {
```

Comment:
There are a few places in the builder where we explicitly set the deposit contract even though it is already set in the constants.yaml.  We should be able to cut these now.  For example [here](https://github.com/ConsenSys/teku/blob/e324db83ef0b009d709fdde280e4649ec5b887c0/ethereum/networks/src/main/java/tech/pegasys/teku/networks/Eth2NetworkConfiguration.java#L270).